processing-tests/ksp-tests/src/test/kotlin/com/livefront/sealedenum/compilation/ksp/NestedObjectsWithSameNameTests.kt

```diff
@@ -0,0 +1,58 @@
+package com.livefront.sealedenum.compilation.ksp
+
+import com.livefront.sealedenum.testing.assertCompiles
+import com.livefront.sealedenum.testing.assertGeneratedFileMatches
+import com.livefront.sealedenum.testing.compile
+import com.livefront.sealedenum.testing.getSourceFile
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+/**
+ * Verifies that a sealed class in a companion object (or more precisely, an annotation that appears on a nested object
+ * with the same name) still generates a sealed enum.
+ *
+ * Note that this only works with KSP, due to a limitation with KAPT.
+ * Because the generated stubs for such an object wouldn't be valid Java, they are skipped, meaning that the annotation
```

Comment:
Because of the nesting or because of specifically `companion`?

Reply:
Because of the nesting!

The bit that ends up not being valid in the stubs is a class being nested in another class where both classes share the same name.

`companion object`s (with their default name of `Companion`) just end up being the most likely case where this will occur with `@GenSealedEnum`, especially since `@GenSealedEnum` has to be placed on a `companion object`.